src/packages/utils/src/things/json-rpc/json-rpc-quantity.ts

```diff
@@ -1,87 +1,132 @@
-import { bufferToBigInt, BUFFER_EMPTY } from "../../utils";
+import { bufferToBigInt } from "../../utils/buffer-to-bigint";
 import { BaseJsonRpcType } from "./json-rpc-base-types";
-// TODO(perf): rewrite this stuff since it isn't really caching anything
+import { JsonRpcInputArg } from "./input-parsers";
+
+const BUFFER_EMPTY = Buffer.alloc(0);
+
 export class Quantity extends BaseJsonRpcType {
+  private static ZERO_VALUE_STRING = "0x0";
+
   _nullable: boolean = false;
-  public static from(
-    value: number | bigint | string | Buffer,
-    nullable = false
-  ) {
+
+  public static from(value: JsonRpcInputArg, nullable = false) {
     if (value instanceof Quantity) return value;
-    const q = new Quantity(value);
-    q._nullable = nullable;
-    return q;
+    return new Quantity(value, nullable);
+  }
+
+  constructor(value: JsonRpcInputArg, nullable?: boolean) {
+    super(value);
+    if (typeof(value) === "string" && value === "0x") {
```

Comment:
Is the following faster?

```suggestion
    if (value === "0x") {
```

Reply:
Note: you can put this check before the `super` call if you so desired:

```
    if (typeof(value) === "string" && value === "0x") {
       //       ...
    }
    super(value);
```

Reply:
Definitely seems unnecessary to check the type _and_ value 🤷 

Regarding the super call - nice! I thought that super() needed to be the first statement in the constructor, I've learned it's a little more nuanced than that:
>A 'super' call must be the first statement in the constructor to refer to 'super' or 'this' when a derived class contains initialized properties, parameter properties, or private identifiers.

Reply:
Oh, this was changed in Typescript 4.6: https://github.com/microsoft/TypeScript/commit/b7fee7f1ee78d05c793409069ddd00105eddad26

We are on 4.5.2, so could consider a major bump. What's your thoughts?

Reply:
We're on 4.6.4 now :-)